metrics/cli/docs/04_managing_firestore_data.md

```diff
@@ -112,3 +120,35 @@ Cons:
 ### Decision
 
 As we've analyzed above, the [Cloud Functions](#using-cloud-functions-for-firebase) does not allow us to automate the Cloud Firestore data managing, so we should choose the [Dart Package](#using-dart-package) since it provides a more clean and fully automated way of managing Cloud Firestore data.
+
+## Design
+
+The following sections provide an implementation of managing Firestore data integration into the Metrics CLI tool. 
+
+Let's take a look on the main classes the managing Firestore data integration requires.
+
+### FirestoreConfigRepository
+
+The `FirestoreConfigRepository` is a repository, which provides methods for managing the [feature config](https://github.com/platform-platform/monorepo/blob/master/docs/19_security_audit_document.md#the-feature_config-collection) 
+and [allowed domains](https://github.com/platform-platform/monorepo/blob/master/docs/19_security_audit_document.md#the-allowed_email_domains-collection) collections data. 
+To implement these methods, the repository using the [dartbase_admin package](https://pub.dev/packages/dartbase_admin) as described in the [decision](#decision).
+
+### FirebaseService
+
+The `FirebaseService` is an existing service interface, in which we should add the following methods required for this integration:
+- `seedFirestoreData` - seeds Firestore database with required data;
+- `initAdminFirestore` - initializes Firestore database with Admin privileges.
+
+### GcloudService
+
+The `GcloudService` is an existing service interface, in which we should add the following methods required for this integration:
+- `downloadServiceKey` - downloads a service account's key;
+- `deactivateServiceKey` - deactivates a service account's key.
```

Comment:
Should we add any methods to the `GcloudCli `class to implement these methods in the `GcloudCliServiceAdapter`?